src/fs.rs

```diff
@@ -295,6 +296,9 @@ pub trait DavFile: Debug + Send + Sync {
     fn read_bytes(&mut self, count: usize) -> FsFuture<bytes::Bytes>;
     fn seek(&mut self, pos: SeekFrom) -> FsFuture<u64>;
     fn flush(&mut self) -> FsFuture<()>;
+    fn redirect_url(&mut self) -> FsFuture<Url> {
```

Comment:
```suggestion
    fn redirect_url(&mut self) -> FsFuture<Option<String>> {
```
改成返回 `Option<String>` 吧，这样可以默认返回 `None`，在 handle_gethead 中对 `None` 的情况就继续往下走。